dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileHandler.java

```diff
@@ -0,0 +1,146 @@
+  public void startDocument() {
+    if (matchers != null) {
+      throw new IllegalStateException("This handler started reading document already");
+    }
+    matchers = ImmutableList.builder();
+  }
+
+  /** Adds {@code child} to the parent matcher that is at the top of the {@link #stack}. */
+  private void addMatcherToParent(SymbolProblemTargetMatcher child) throws SAXException {
+    SymbolProblemMatcher parent = stack.peek();
+    if (parent instanceof SourceMatcher && child instanceof SymbolProblemSourceMatcher) {
+      ((SourceMatcher) parent).setMatcher((SymbolProblemSourceMatcher) child);
+    } else if (parent instanceof TargetMatcher) {
+      ((TargetMatcher) parent).setMatcher(child);
+    } else {
+      throw new SAXException(
+          "Unexpected parent-child relationship. Parent:" + parent + ", child:" + child);
+    }
+  }
+
+  @Override
+  public void startElement(
+      String namespaceUri, String localName, String qualifiedName, Attributes attributes)
+      throws SAXException {
+    if (!namespaceUri.isEmpty()) {
+      throw new SAXException(
+          "unrecognized element: " + qualifiedName + " in namespace " + namespaceUri);
+    }
+    switch (localName) {
+      case "LinkageCheckerFilter":
+        break;
+      case "LinkageError":
+        stack.push(new LinkageErrorMatcher());
+        break;
+      case "Source":
+        SourceMatcher sourceMatcher = new SourceMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setSourceMatcher(sourceMatcher);
+        stack.push(sourceMatcher);
+        break;
+      case "Target":
+        TargetMatcher targetMatcher = new TargetMatcher();
+        ((LinkageErrorMatcher) stack.peek()).setTargetMatcher(targetMatcher);
+        stack.push(targetMatcher);
+        break;
+      case "Package":
+        PackageMatcher packageMatcher = new PackageMatcher(attributes.getValue("name"));
+        addMatcherToParent(packageMatcher);
+        break;
+      case "Class":
+        String classNameOnClass = attributes.getValue("name");
+        ClassMatcher classMatcher = new ClassMatcher(classNameOnClass);
+        addMatcherToParent(classMatcher);
+        break;
+      case "Method":
+        String classNameOnMethod = attributes.getValue("className");
+        MethodMatcher methodMatcher =
+            new MethodMatcher(classNameOnMethod, attributes.getValue("name"));
+        addMatcherToParent(methodMatcher);
+        break;
+      case "Field":
+        String classNameOnField = attributes.getValue("className");
+        FieldMatcher fieldMatcher = new FieldMatcher(classNameOnField, attributes.getValue("name"));
+        addMatcherToParent(fieldMatcher);
+        break;
+      default:
+        throw new SAXException("Unknown tag " + localName);
+    }
+  }
+
+  @Override
+  public void endElement(String namespaceUri, String localName, String qualifiedName)
+      throws SAXException {
+    SymbolProblemMatcher poppedMatcher;
+    if (!namespaceUri.isEmpty()) {
```

Comment:
I don't think this case can happen, given that this case would have been detected already unless the document is malformed, in which case you'll also have errored out before this is called.

Reply:
Removed this check in endElement.